chall04/fulldemo.py

```diff
@@ -0,0 +1,80 @@
+#! /usr/local/bin/python3
+
+import sys
+import fileinput
+
+def error(e, n):
+	if e == 0:
+		if n != 0:
+			print("%s: %s: Can't read file" % (sys.argv[0], sys.argv[n]))
+		else:
+			print("%s: stdin: Can't read file" % (sys.argv[0]))
+	else:
+		if n != 0:
+			print("%s: %s: Not enough space in the given shelves" % (sys.argv[0], sys.argv[n]))
+		else:
+			print("%s: stdin: Not enough space in the given shelves" % (sys.argv[0]))
+
+def shelves_to_use(all_file):
+	shelves_use = 0
+	for i in range(1, len(all_file)):
+		line = all_file[i].split()
+		if len(line) < 2:
+			return (-1)
+		try:
+			shelves_use = int(line[0]) + shelves_use
+			if int(line[0]) < 0:
+				return (-1)
+		except:
+			return (-1)
+	return(shelves_use)
+
+def	check_to_buy(array):
+	for i in range(len(array)):
+		try:
+			if int(array[i]) < 0:
+				return (-1)
+		except:
+			return (-1)
```

Comment:
Why would you return `-1`? You should return either `True` or `False`.
Also there is a more pythonic way to do this
```suggestion
	try:
		return all(i >= 0 for i in array)
	except:
		return False
```